src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2624,7 +2624,7 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
             if (used)
             {
                 // default type parameter values must be emitted as 'initobj' regardless of constraints
-                if (!type.IsTypeParameter())
+                if (!type.IsTypeParameter() && type.SpecialType != SpecialType.System_Decimal)
```

Comment:
It appears that VB does weird things with regards to emitting Decimal and DateTime constants.

http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/CodeGen/EmitExpression.vb,39